x/gov/keeper_test.go

```diff
@@ -0,0 +1,205 @@
+
+	votesIterator := keeper.GetVotes(ctx, proposalID)
+	assert.True(t, votesIterator.Valid())
+	keeper.cdc.MustUnmarshalBinary(votesIterator.Value(), vote)
+	assert.True(t, votesIterator.Valid())
+	assert.Equal(t, addrs[0], vote.Voter)
+	assert.Equal(t, proposalID, vote.ProposalID)
+	assert.Equal(t, "Yes", vote.Option)
+	votesIterator.Next()
+	assert.True(t, votesIterator.Valid())
+	keeper.cdc.MustUnmarshalBinary(votesIterator.Value(), vote)
+	assert.True(t, votesIterator.Valid())
+	assert.Equal(t, addrs[1], vote.Voter)
+	assert.Equal(t, proposalID, vote.ProposalID)
+	assert.Equal(t, "NoWithVeto", vote.Option)
+	votesIterator.Next()
+	assert.False(t, votesIterator.Valid())
+}
+
+func TestProposalQueues(t *testing.T) {
+	ctx, _, keeper := createTestInput(t, false, 100)
+
+	assert.Nil(t, keeper.InactiveProposalQueuePeek(ctx))
+	assert.Nil(t, keeper.ActiveProposalQueuePeek(ctx))
+
+	proposal := keeper.NewProposal(ctx, "Test", "description", "Text")
+	proposal2 := keeper.NewProposal(ctx, "Test2", "description", "Text")
+	proposal3 := keeper.NewProposal(ctx, "Test3", "description", "Text")
+	proposal4 := keeper.NewProposal(ctx, "Test4", "description", "Text")
+
+	keeper.InactiveProposalQueuePush(ctx, proposal)
+	keeper.InactiveProposalQueuePush(ctx, proposal2)
+	keeper.InactiveProposalQueuePush(ctx, proposal3)
+	keeper.InactiveProposalQueuePush(ctx, proposal4)
+
+	assert.Equal(t, keeper.InactiveProposalQueuePeek(ctx).ProposalID, proposal.ProposalID)
+	assert.Equal(t, keeper.InactiveProposalQueuePop(ctx).ProposalID, proposal.ProposalID)
+	assert.Equal(t, keeper.InactiveProposalQueuePeek(ctx).ProposalID, proposal2.ProposalID)
+	assert.Equal(t, keeper.InactiveProposalQueuePop(ctx).ProposalID, proposal2.ProposalID)
+	assert.Equal(t, keeper.InactiveProposalQueuePeek(ctx).ProposalID, proposal3.ProposalID)
+	assert.Equal(t, keeper.InactiveProposalQueuePop(ctx).ProposalID, proposal3.ProposalID)
+	assert.Equal(t, keeper.InactiveProposalQueuePeek(ctx).ProposalID, proposal4.ProposalID)
+	assert.Equal(t, keeper.InactiveProposalQueuePop(ctx).ProposalID, proposal4.ProposalID)
+
+	keeper.ActiveProposalQueuePush(ctx, proposal)
+	keeper.ActiveProposalQueuePush(ctx, proposal2)
+	keeper.ActiveProposalQueuePush(ctx, proposal3)
+	keeper.ActiveProposalQueuePush(ctx, proposal4)
+
+	assert.Equal(t, keeper.ActiveProposalQueuePeek(ctx).ProposalID, proposal.ProposalID)
+	assert.Equal(t, keeper.ActiveProposalQueuePop(ctx).ProposalID, proposal.ProposalID)
+	assert.Equal(t, keeper.ActiveProposalQueuePeek(ctx).ProposalID, proposal2.ProposalID)
+	assert.Equal(t, keeper.ActiveProposalQueuePop(ctx).ProposalID, proposal2.ProposalID)
+	assert.Equal(t, keeper.ActiveProposalQueuePeek(ctx).ProposalID, proposal3.ProposalID)
+	assert.Equal(t, keeper.ActiveProposalQueuePop(ctx).ProposalID, proposal3.ProposalID)
+	assert.Equal(t, keeper.ActiveProposalQueuePeek(ctx).ProposalID, proposal4.ProposalID)
+	assert.Equal(t, keeper.ActiveProposalQueuePop(ctx).ProposalID, proposal4.ProposalID)
```

Comment:
same for the above two test functions - small code comments throughout would be useful

Reply:
Also can this be abstracted into something that takes an array of proposals? (May be useful for randomized testing suite, plus this has lots of reuse)